src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportFromFileService.java

```diff
@@ -0,0 +1,193 @@
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.Repository;
+import de.tum.in.www1.artemis.domain.VcsRepositoryUrl;
+import de.tum.in.www1.artemis.domain.enumeration.RepositoryType;
+import de.tum.in.www1.artemis.service.FileService;
+import de.tum.in.www1.artemis.service.RepositoryService;
+import de.tum.in.www1.artemis.service.StaticCodeAnalysisService;
+import de.tum.in.www1.artemis.service.ZipFileService;
+import de.tum.in.www1.artemis.service.connectors.GitService;
+import de.tum.in.www1.artemis.web.rest.errors.BadRequestAlertException;
+
+@Service
+public class ProgrammingExerciseImportFromFileService {
+
+    private final ProgrammingExerciseService programmingExerciseService;
+
+    private final ZipFileService zipFileService;
+
+    private final StaticCodeAnalysisService staticCodeAnalysisService;
+
+    private final RepositoryService repositoryService;
+
+    private final GitService gitService;
+
+    private final FileService fileService;
+
+    public ProgrammingExerciseImportFromFileService(ProgrammingExerciseService programmingExerciseService, ZipFileService zipFileService,
+            StaticCodeAnalysisService staticCodeAnalysisService, RepositoryService repositoryService, GitService gitService, FileService fileService) {
+        this.programmingExerciseService = programmingExerciseService;
+        this.zipFileService = zipFileService;
+        this.staticCodeAnalysisService = staticCodeAnalysisService;
+        this.repositoryService = repositoryService;
+        this.gitService = gitService;
+        this.fileService = fileService;
+    }
+
+    /**
+     * Imports a programming exercise from an uploaded zip file that has previously been downloaded from an Artemis instance.
+     * It first extracts the contents of the zip file, then creates a programming exercise (same process as creating a new one),
+     * then deletes the template content initially pushed to the repositories and copies over the extracted content
+     *
+     * @param programmingExerciseForImport the programming exercise that should be imported
+     * @param zipFile                      the zip file that contains the exercise
+     * @param course                       the course to which the exercise should be added
+     * @return the imported programming exercise
+     **/
+    public ProgrammingExercise importProgrammingExerciseFromFile(ProgrammingExercise programmingExerciseForImport, MultipartFile zipFile, Course course)
+            throws IOException, GitAPIException, URISyntaxException {
+        Path importExerciseDir = Files.createTempDirectory("imported-exercise-dir");
+        Path exerciseFilePath = Files.createTempFile(importExerciseDir, "exercise-for-import", ".zip");
+        if (!"zip".equals(FileNameUtils.getExtension(zipFile.getOriginalFilename()))) {
+            throw new BadRequestAlertException("The file is not a zip file", "programmingExercise", "fileNotZip");
+        }
```

Comment:
```suggestion
        if (!"zip".equals(FileNameUtils.getExtension(zipFile.getOriginalFilename()))) {
            throw new BadRequestAlertException("The file is not a zip file", "programmingExercise", "fileNotZip");
        }

        Path importExerciseDir = Files.createTempDirectory("imported-exercise-dir");
        Path exerciseFilePath = Files.createTempFile(importExerciseDir, "exercise-for-import", ".zip");
```
The check can happen first before any filesystem operations are made.